### DISCUSSION TITLE
Concurrency in ngs?

### DISCUSSION THREAD

**Discussion by MinmoTech** [ngs-lang/ngs#606] [Q&A] [answered]
Is there a way to do concurrency or parallelism in ngs yet?
The only info I've found yet is this issue: https://github.com/ngs-lang/ngs/issues/139

Right now, even being able to fork a function would be enough for me, but more control over it would be awesome of course :)

**Comment by ilyash-b** [comment, marked as answer]
You can already do concurrency:
* `Thread()` - https://ngs-lang.org/doc/latest/generated/Thread.html
* `pmap()` - https://ngs-lang.org/doc/latest/generated/pmap.html
* `ptimes()` - https://ngs-lang.org/doc/latest/generated/ptimes.html
* `process=$(blah &)` and then `process.join()`

The issue you mentioned is specifically about thread-safe data structures
